src/chunk.c

```diff
@@ -1099,21 +1099,47 @@ chunk_create_from_hypercube_and_table_after_lock(const Hypertable *ht, Hypercube
 	chunk->hypertable_relid = ht->main_table_relid;
 	Assert(OidIsValid(ht->main_table_relid));
 
+	Relation chunk_rel = table_open(chunk_table_relid, AccessExclusiveLock);
+	Relation ht_rel = table_open(ht->main_table_relid, AccessExclusiveLock);
+	TupleDesc tupdesc = RelationGetDescr(chunk_rel);
+
+	for (int attno = 0; attno < tupdesc->natts; attno++)
+	{
+		Form_pg_attribute att = TupleDescAttr(tupdesc, attno);
+		char *attName = NameStr(att->attname);
+
+		/* Ignore dropped */
+		if (att->attisdropped)
+			continue;
+
+		/* Try to find the column in parent (matching on column name) */
+		if (!SearchSysCacheExists2(ATTNAME,
+								   ObjectIdGetDatum(RelationGetRelid(ht_rel)),
+								   CStringGetDatum(attName)))
```

Comment:
You could use the Name type directly:

```suggestion
								   NameGetDatum(&att->attname)))
```